src/prover.rs

```diff
@@ -1382,27 +1488,30 @@ pub fn compute_evaluations_over_lagrange_basis<'a, A: GoodAllocator>(
     // additionally, copy permutation polynomial should be opened at "z*w"
     // lookup polynomials should be opened at "0"
     // we should follow order of the evaluations as in the reference impl
-    let trace_storage = trace_holder.get_or_compute_coset_evals(0)?;
-    let setup_storage = setup_holder.get_or_compute_coset_evals(0)?;
-    let domain_size = trace_storage.domain_size();
+    let domain_size = setup_holder.domain_size;
     let precomputed_bases_for_z = PrecomputedBasisForBarycentric::precompute(domain_size, z)?;
-
-    let trace_evals_at_z = trace_storage.barycentric_evaluate::<A>(&precomputed_bases_for_z)?;
-    let setup_evals_at_z = setup_storage.barycentric_evaluate::<A>(&precomputed_bases_for_z)?;
-
-    let quotient_storage = quotient_holder.get_or_compute_coset_evals(0)?;
-    let quotient_evals_at_z = quotient_storage.barycentric_evaluate(&precomputed_bases_for_z)?;
+    let trace_evals_at_z = {
+        let trace_storage = trace_holder.get_coset_evaluations(0)?;
+        trace_storage.barycentric_evaluate::<A>(&precomputed_bases_for_z)?
+    };
+    let setup_evals_at_z = {
+        let setup_storage = setup_holder.get_coset_evaluations(0)?;
+        setup_storage.barycentric_evaluate::<A>(&precomputed_bases_for_z)?
+    };
+    let quotient_evals_at_z = {
+        let quotient_storage = quotient_holder.get_or_compute_coset_evals(0)?;
+        quotient_storage.barycentric_evaluate(&precomputed_bases_for_z)?
+    };
```

Comment:
Looks like you streamlined the temporaries here a lot. Were these temporaries the biggest issue with barycentric memory high water mark? Do you think I should still investigate the boojum-cuda side?

Reply:
the barycentric evaluation still needs a good chunk of temporary allocations but we don't need to invest more time to try to make them smaller, it is fine as it is